src/priv_appendable_data.rs

```diff
@@ -0,0 +1,418 @@
+// Copyright 2015 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net Commercial License,
+// version 1.0 or later, or (2) The General Public License (GPL), version 3, depending on which
+// licence you accepted on initial access to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project generally, you agree to be
+// bound by the terms of the MaidSafe Contributor Agreement, version 1.0.  This, along with the
+// Licenses can be found in the root directory of this project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.
+//
+// Please review the Licences for the specific language governing permissions and limitations
+// relating to use of the SAFE Network Software.
+
+use maidsafe_utilities::serialisation::serialise;
+use rust_sodium::crypto::sign::{self, PublicKey, SecretKey, Signature};
+use std::collections::BTreeSet;
+use std::fmt::{self, Debug, Formatter};
+use xor_name::XorName;
+use data::DataIdentifier;
+use error::RoutingError;
+use pub_appendable_data::{SERIALISED_APPENDED_DATA_SIZE, AppendedData, Filter};
+
+/// Maximum allowed size for a private appendable data to grow to
+pub const MAX_PRIV_APPENDABLE_DATA_SIZE_IN_BYTES: usize = 102400;
+
+/// Private appendable data.
+///
+/// These types may be stored unsigned with previous and current owner keys
+/// set to the same keys. Updates require a signature to validate.
+///
+/// Data can be appended by any key that is not excluded by the filter.
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, RustcDecodable, RustcEncodable)]
+pub struct PrivAppendableData {
+    name: XorName,
+    version: u64,
+    current_owner_keys: Vec<PublicKey>,
+    previous_owner_keys: Vec<PublicKey>,
+    filter: Filter,
+    deleted_data: BTreeSet<AppendedData>,
+    previous_owner_signatures: Vec<Signature>, // All the above fields
+    data: BTreeSet<AppendedData>, // Unsigned
```

Comment:
This should probably be `PrivAppendedData` - and that has probably also a different serialised size.